### PR TITLE
Add errors.send() helper function to prevent "cannot set headers" error.

### LINKED ISSUE #259: Cannot set headers after they are sent...again.
Promises are eternally problematic, it seems. In two places (times.js:267 and times.js:322), rejecting promises while sending errors causes multiple errors to try to be set. Perhaps better to just reject with the error object.

### PR DESCRIPTION
**Pull request by LyonesGamer**
Fixes #259 by checking whether data has already been sent before sending any errors.

### CODE REVIEW

File: src/errors.js [hunk 1]
```diff
@@ -202,4 +202,15 @@ module.exports = {
     return createError(400, 'Bad Query Value', 'Parameter ' + key +
     ' contained invalid value ' + value);
   },
+
+  /*
+   * Helper function to ensure HTTP status is sent and prevent repeated send
+   */
+  send: function(error, res) {
+    if (!res.finished) {
```
Comment (tschuy) [inline src/errors.js]: Sorry, commenting from a point of ignorance of how errors are currently handled. Are errors that occur but aren't able to be sent still recorded in a log, hopefully with some big warning saying, hey, shit happened?

Comment (LyonesGamer) [inline src/errors.js]: If they're a server error, yes. And almost all the time that we get this error, the second error is pretty irrelevant anyway.

**Comment by ElijahCaine** [comment]
Tests pass, code looks good, great idea I only wish we implemented it soooner. :+1:

**Comment by Kennric** [comment]
I don't see a problem with this, but wish we had a clear test case for the problem. +1

**Pull request merged**